test/bin/scenario.sh

```diff
@@ -38,6 +38,48 @@ full_vm_name() {
     echo "${SCENARIO//@/-}-${base}"
 }
 
+# hostname validation
+# based on https://github.com/rhinstaller/anaconda/blob/c95142f76735a2e9ae6d845f8569d46632ddd619/pyanaconda/network.py#L96-L120
+validate_full_vm_name() {
+    local vm_name="$1"
+
+    if [ -z "${vm_name}" ]; then
+        error "VM hostname cannot be empty string" 
+        record_junit "${vmname}" "vm_name_validation" "FAILED"
+        exit 1
+    fi
+
+    if [ ${#vm_name} -gt 64 ]; then
+        error "VM hostname is too long" 
+        record_junit "${vmname}" "vm_name_validation" "FAILED"
+        exit 1
+    fi
+
+    if [[ $vm_name =~ ^\. ]]; then
+        error "VM hostname cannot start with '.'" 
+        record_junit "${vmname}" "vm_name_validation" "FAILED"
+        exit 1
+    fi
+
+    # split by '.' and check every section separately
+    IFS='.' read -ra ARR <<< "$vm_name"
```

Comment:
If we validate `$vm_hostname` instead of `$full_vmname`, then we can remove this loop and flatten and merge the checks.